Makefile

```diff
@@ -70,7 +70,7 @@ EXTRACTED_DIR := extracted/$(VERSION)
 VENV := .venv
 
 MAKE = make
-CPPFLAGS += -fno-dollars-in-identifiers -P
+CPPFLAGS += -P -xc++ -fno-dollars-in-identifiers
```

Comment:
This is its own rabbit hole unfortunately but
Why `c++` instead of e.g. `c`
I also see `assembler` and `assembler-with-cpp` are options

In any case probably -x should be part of rules (which know which file types they deal with) instead of the base cpp command?

Reply:
I copied this from the previous macOS behavior but didn't investigate. Presumably for `//` comments? Maybe it was the macOS `cpp` being weird again, recent gcc/clang versions should default to more recent C standards anyway.

I changed this to `c` since this is probably what we mean when we say to "run the C preprocessor on this file". I strongly think all of our non-C cpp usage should have the same behavior, using different behavior for different files sounds confusing.

Reply:
Which files do we use cpp on?

Reply:
`spec`, `undefined_syms.txt`, `assets/text/%.h`, `%.s`

Arguably the last one should use `mips-linux-gnu-gcc` with a `.S` file instead (or `-xassembler-with-cpp`), but I don't think we should require that dependency. I can't find any docs on the difference between the preprocessing for `assembler-with-cpp` and `c`, which makes me not want to mess with it